src/updater/updater.cpp

```diff
@@ -149,446 +95,568 @@ void CheckWeightsConsistency(
+        for (auto geometry_id = range.begin(); geometry_id < range.end(); geometry_id++)
         {
-            const file_mapping mapping{filename.c_str(), mode};
-            mapped_region region{mapping, mode};
-            region.advise(mapped_region::advice_sequential);
-            return region;
+            auto nodes_range = segment_data.GetForwardGeometry(geometry_id);
+
+            segment_lengths.clear();
+            segment_lengths.reserve(nodes_range.size() + 1);
+            util::for_each_pair(nodes_range, [&](const auto &u, const auto &v) {
+                segment_lengths.push_back(util::coordinate_calculation::greatCircleDistance(
+                    util::Coordinate{internal_to_external_node_map[u]},
+                    util::Coordinate{internal_to_external_node_map[v]}));
+            });
+
+            auto fwd_weights_range = segment_data.GetForwardWeights(geometry_id);
+            auto fwd_durations_range = segment_data.GetForwardDurations(geometry_id);
+            auto fwd_datasources_range = segment_data.GetForwardDatasources(geometry_id);
+            bool fwd_was_updated = false;
+            for (auto segment_offset = 0UL; segment_offset < fwd_weights_range.size();
+                 ++segment_offset)
+            {
+                auto u = internal_to_external_node_map[nodes_range[segment_offset]].node_id;
+                auto v = internal_to_external_node_map[nodes_range[segment_offset + 1]].node_id;
+                if (auto value = segment_speed_lookup({u, v}))
+                {
+                    auto new_duration =
+                        convertToDuration(value->speed, segment_lengths[segment_offset]);
+                    auto new_weight =
+                        convertToWeight(value->weight, weight_multiplier, new_duration);
+                    fwd_was_updated = true;
+
+                    fwd_weights_range[segment_offset] = new_weight;
+                    fwd_durations_range[segment_offset] = new_duration;
+                    fwd_datasources_range[segment_offset] = value->source;
+                    counters[value->source] += 1;
+                }
+                else
+                {
+                    counters[LUA_SOURCE] += 1;
+                }
+            }
+            if (fwd_was_updated)
+                updated_segments.push_back(GeometryID{geometry_id, true});
+
+            // In this case we want it oriented from in forward directions
+            auto rev_weights_range =
+                boost::adaptors::reverse(segment_data.GetReverseWeights(geometry_id));
+            auto rev_durations_range =
+                boost::adaptors::reverse(segment_data.GetReverseDurations(geometry_id));
+            auto rev_datasources_range =
+                boost::adaptors::reverse(segment_data.GetReverseDatasources(geometry_id));
+            bool rev_was_updated = false;
+
+            for (auto segment_offset = 0UL; segment_offset < fwd_weights_range.size();
```

Comment:
Check all 0UL's - either replace with 0ull or better size_t instead of auto.

0ul is unsigned long; longs are only guaranteed to have 32 bits (in reality: Linux 64 bit, Windows 32 bit).

http://en.cppreference.com/w/cpp/language/types